realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -0,0 +1,423 @@
+            @Override
+            public void onChange(AllJavaTypes object) {
+                counter.incrementAndGet();
+                looperThread.testComplete();
+            }
+        };
+        gen1.addChangeListener(listener);
+
+        ltRealm.beginTransaction();
+
+        AllJavaTypes gen2A = ltRealm.createObject(AllJavaTypes.class, 1);
+        gen2A.setFieldObject(gen1);
+        ltRealm.commitTransaction();
+
+        assertEquals(1, counter.get());
+        assertEquals(2, ltRealm.where(AllJavaTypes.class).findAll().size());
+    }
+
+    // A notification callback should be called when a backlinked object is deleted
+    @Test
+    @RunTestInLooperThread
+    public void notificationOnDelete() {
+        Realm ltRealm = looperThread.realm;
+
+        ltRealm.beginTransaction();
+        AllJavaTypes gen1 = ltRealm.createObject(AllJavaTypes.class, 10);
+        AllJavaTypes gen2A = ltRealm.createObject(AllJavaTypes.class, 1);
+        gen2A.setFieldObject(gen1);
+        ltRealm.commitTransaction();
+
+        final AtomicInteger counter = new AtomicInteger(0);
+        RealmChangeListener<AllJavaTypes> listener = new RealmChangeListener<AllJavaTypes>() {
+            @Override
+            public void onChange(AllJavaTypes object) {
+                counter.incrementAndGet();
+                looperThread.testComplete();
+            }
+        };
+        gen1.addChangeListener(listener);
+
+        ltRealm.beginTransaction();
+        ltRealm.where(AllJavaTypes.class).equalTo("fieldId", 1).findAll().deleteAllFromRealm();
+        ltRealm.commitTransaction();
+
+        assertEquals(1, ltRealm.where(AllJavaTypes.class).findAll().size());
+        assertEquals(1, counter.get());
+    }
+
+    // A notification callback should not be called for unrelated changes on the same object
+    @Test
+    @RunTestInLooperThread
+    public void notificationNotSentOnUnrelatedChange() {
+        Realm ltRealm = looperThread.realm;
+
+        ltRealm.beginTransaction();
+        AllJavaTypes gen1 = ltRealm.createObject(AllJavaTypes.class, 10);
+        AllJavaTypes gen2A = ltRealm.createObject(AllJavaTypes.class, 1);
+        ltRealm.commitTransaction();
+
+        final AtomicInteger counter = new AtomicInteger(0);
+        RealmChangeListener<AllJavaTypes> listener = new RealmChangeListener<AllJavaTypes>() {
+            @Override
+            public void onChange(AllJavaTypes object) {
+                counter.incrementAndGet();
+                looperThread.testComplete();
+            }
+        };
+        gen1.addChangeListener(listener);
+
+        ltRealm.beginTransaction();
+        ltRealm.where(AllJavaTypes.class).equalTo("fieldId", 1).findAll().deleteAllFromRealm();
+        ltRealm.commitTransaction();
+
+        assertEquals(1, ltRealm.where(AllJavaTypes.class).findAll().size());
+        assertEquals(1, counter.get());
+    }
+
+    // ???
```

Comment:
Well... I think a better idea would be to implement it.  I'm pretty sure that I don't have anything like complete coverage here...